src/NuGet.Clients/NuGet.CommandLine/Commands/Command.cs

```diff
@@ -120,7 +120,12 @@ public void Execute()
                 RepositoryFactory = new CommandLineRepositoryFactory(Console);
                 UserAgent.UserAgentString = UserAgent.CreateUserAgentString(CommandLineConstants.UserAgent);
 
-                ExecuteCommandAsync().Wait();
+                // Starts the asynchronous task on a thread-pool thread.
+                // Returns a proxy to the original task.
+                var task = Task.Run(() => ExecuteCommandAsync());
+
+                // Will block until the task is completed...
+                task.Wait();
             }
```

Comment:
This is the magic that sends async task to a threadpool while the main thread is blocked on waiting.